### PR TITLE
fix: Focus handling and shortcut key for filters

### PR DESCRIPTION
**Pull request by steinsebastian**

### CODE REVIEW

File: feature-libs/product-configurator/rulebased/components/overview-filter-bar/configurator-overview-filter-bar.component.spec.ts [hunk 1]
```diff
@@ -300,5 +324,127 @@ describe('ConfiguratorOverviewFilterBarComponent', () => {
...
+        const result = component['getNextElementIdToFocusForAttributeFilter'](
+          0,
+          attrFilters
+        );
+        expect(result).toBeNull();
+      });
+    });
+
+    describe('getNextElementIdToFocusForGroupFilter', () => {
+      it('should return the next element id when indexOfRemoved is less than the length of groupFilters', () => {
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/overview-filter-bar/configurator-overview-filter-bar.component.spec.ts]:
```suggestion
      it('should return the next element id if indexOfRemoved is less than the length of groupFilters', () => {
```
Please change other test descriptions as well

File: feature-libs/product-configurator/rulebased/components/overview-filter-bar/configurator-overview-filter-bar.component.ts [hunk 2]
```diff
@@ -70,15 +96,96 @@ export class ConfiguratorOverviewFilterBarComponent {
...
     this.configuratorCommonsService.updateConfigurationOverview(
       this.createInputConfig(config, attrFilters, groupFilters)
     );
   }
 
+  /**
+   * Determines the next element to focus on after removing an attribute filter.
+   * Return null if removed attribute filter was the last one in the list.
+   *
+   * @param {number} indexOfRemoved - The index of the attribute filter that has been removed.
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/overview-filter-bar/configurator-overview-filter-bar.component.ts]: Please omit the type specification. As far as I recall this is what we agreed upon. Please do so for other JSDocs 
```suggestion
   * @param  indexOfRemoved - The index of the attribute filter that has been removed.
```

**Review by ChristophHi — CHANGES_REQUESTED** [review]

File: feature-libs/product-configurator/rulebased/components/overview-filter-bar/configurator-overview-filter-bar.component.spec.ts (hunk 1 above)
Comment (steinsebastian) [reply feature-libs/product-configurator/rulebased/components/overview-filter-bar/configurator-overview-filter-bar.component.spec.ts]: done

File: feature-libs/product-configurator/rulebased/components/overview-filter-bar/configurator-overview-filter-bar.component.ts (hunk 2 above)
Comment (steinsebastian) [reply feature-libs/product-configurator/rulebased/components/overview-filter-bar/configurator-overview-filter-bar.component.ts]: done

**Review by ChristophHi — APPROVED** [review]

**Pull request merged**